docs/src/references/mqtt-topic-guidelines.md

```diff
@@ -0,0 +1,165 @@
+   in MQTT topics over their globally unique cloud IDs (typically very long and cumbersome) would be desired.
+   Ideally, all translation between user-provided-ids and cloud-twin-ids should be done internally by thin-edge,
+   as long as it doesn't adversely affect any data mapping cost at scale.
+
+### Out of scope
+
+1. Routing different kinds of data to different clouds, e.g: all telemetry to azure and all commands from/to Cumulocity.
+   Even though this requirement is realistic, thin-edge MQTT topics must not be corrupted with cloud specific semantics,
+   and the same requirement will be handled with some external routing mechanism(e.g: routing via bridge topics)
+1. Services on child devices. Service support can be limited to thin-edge devices only.
+
+## Proposals
+
+### Dedicated topics for tedge device, services and child devices
+
+The topics for the thin-edge device, the services running on it and child devices have different prefixes:
+
+For parent: tedge/main/
+For services: tedge/service/<service-id>
+For immediate child devices: tedge/child/<child-id>
+For nested child devices: tedge/child/<child-id>
+
+#### Telemetry
+
+For telemetry data, the topics would be grouped under a `telemetry/` sub-topic as follows:
+* Measurements: `tedge/main/telemetry/measurements`, `tedge/child/<child-id>/telemetry/measurements` etc
+* Events: `tedge/main/telemetry/events/<event-type>`, `tedge/service/<service-id>/telemetry/events/<event-type>` etc
+* Alarms: `tedge/main/telemetry/alarms/<alarm-type>/<severity>`, `tedge/service/<service-id>/telemetry/<alarm-type>/<severity>` etc
+
+For child devices and services, a similar structure is followed like: `tedge/child/<child-id>/telemetry/measurements`,
+`tedge/service/<service-id>/telemetry/events/<event-type>` etc
+
+#### Commands
+
+Similarly, all commands would be grouped under a `commands/` sub-topic as follows:
+For requests: `tedge/main/commands/req/<operation-type>/<operation-specific-keys>...`
+For responses: `tedge/main/commands/res/<operation-type>/<operation-specific-keys>...`
+
+The `operation-specific-keys` are optional and the number of such keys (topic levels) could vary from one operation to another.
+
+Examples:
+* Software list operation: `tedge/main/commands/req/software_list` and `tedge/main/commands/res/software_list`
+* Software update operation `tedge/main/commands/req/software_update` and `tedge/main/commands/res/software_update`
+* Firmware update operation `tedge/main/commands/req/firmware_update` and `tedge/main/commands/res/firmware_update`
+* Device restart operation `tedge/main/commands/req/device_restart` and `tedge/main/commands/res/device_restart`
+* Configuration snapshot operation `tedge/main/commands/req/config_snapshot` and `tedge/main/commands/res/config_snapshot`
+* Configuration update operation `tedge/main/commands/req/config_update` and `tedge/main/commands/res/config_update`
```

Comment:
I wonder how to make this consistent with the [proposal for MQTT-state-machine](https://github.com/didier-wenzek/tedge-mqtt-state-machine) to handle operations with user provided hooks.